dsa-solutions/lc-solutions/0200-0299/0227-Basic-Calculator-II.md

```diff
@@ -0,0 +1,213 @@
+---
+id: basic-calculator-II
+title: Basic Calculator II
+sidebar_label: 0227 Basic Calculator II
+tags: [Math, String, Stack]
+description: "This gives solution to the problem basic calculator II"
+---
+
+## Problem
+
+Given a string s which represents an expression, evaluate this expression and return its value.
+
+The integer division should truncate toward zero.
+
+You may assume that the given expression is always valid. All intermediate results will be in the range of [-2^31, 2^31 - 1].
+
+Note: You are not allowed to use any built-in function which evaluates strings as mathematical expressions, such as eval().
+
+### Examples
+
+**Example 1:**
+
+**Input:** `s = "3+2*2"`  
+**Output:** `7`
+
+**Example 2:**
+
+**Input:** `s = "3/2"`  
+**Output:** `1`
+
+**Example 3:**
+
+**Input:** `s = "3+5 / 2"`  
+**Output:** `5`
+
+---
+
+### Constraints
+
+- \(1 \leq s.length \leq 3 \times 10^5\)
```

Comment:
replace `\(1 \leq s.length \leq 3 \times 10^5\)` to `$1 \leq s.length \leq 3 \times 10^5$`